config_utilities/test/include/config_utilities/test/utils.h

```diff
@@ -91,6 +91,7 @@ struct RegistrationGuard {
   ~RegistrationGuard() {
     internal::ConfigFactory<BaseT>::template removeEntry<ConfigT>(type);
     internal::ObjectWithConfigFactory<BaseT, Args...>::removeEntry(type);
+    internal::ModuleRegistry::registerConfig<DerivedT, ConfigT>(type);
```

Comment:
Am I missing something for why this has to be re-registered in the destructor? I don't think this changes too much when running the unit tests via ctest, but might break the factory registration tests when running all the tests locally

Reply:
Good catch! I have honestly no idea how that got there 😱